File: LeetCode/Java Solutions/Sum of Digits of String After Convert/sum.java
class Solution {
    public int getLucky(String s, int k) {
        // Step 1: Convert the string into the numeric form
        StringBuilder numStr = new StringBuilder();
        for (char c : s.toCharArray()) {
            numStr.append(c - 'a' + 1);
        }
        
        // Step 2: Perform the digit sum transformation k times
        String result = numStr.toString();
        for (int i = 0; i < k; i++) {
            int sum = 0;
            for (char digit : result.toCharArray()) {
                sum += digit - '0';
            }
            result = String.valueOf(sum);
        }
        
        // Convert the final result back to integer
        return Integer.parseInt(result);
    }
}

File: LeetCode/Python Solutions/Sum of Digits of String After Convert/sum.py
class Solution:
    def getLucky(self, s: str, k: int) -> int:
        # Step 1: Convert the string into the numeric form
        num_str = ''.join(str(ord(char) - ord('a') + 1) for char in s)
        
        # Step 2: Perform the digit sum transformation k times
        for _ in range(k):
            num_str = str(sum(int(digit) for digit in num_str))
        
        # Convert the final result back to integer
        return int(num_str)
